ui-manchette/src/types.ts

```diff
@@ -1,73 +1,24 @@
 import { type CSSProperties } from 'react';
 
-export type PathProperties = {
-  operational_points?:
-    | {
-        extensions?: OperationalPointExtensions;
-        id: string;
-        part: OperationalPointPart;
-        /** Distance from the beginning of the path in mm */
-        position: number;
-      }[]
-    | null;
+export type Waypoint = {
+  id: string;
```

Comment:
Before this patch, there was no unique waypoint ID passed in. The OP ID was passed in, but the same OP could appear multiple times in the path.

I think a unique waypoint ID is a great idea, see https://github.com/OpenRailAssociation/osrd-ui/pull/668#discussion_r1816589010

Reply:
Cool, something to keep in mind when writing the OSRD patch to upgrade to this new API.